src/FS.Embedded/Manifest/ManifestDirectory.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class ManifestDirectory : ManifestEntry
+    {
+        protected ManifestDirectory(string name, ManifestEntry[] children)
+            : base(name)
+        {
+            if (children == null)
+            {
+                throw new ArgumentNullException(nameof(children));
+            }
+
+            Children = children;
+        }
+
+        public ManifestEntry[] Children { get; protected set; }
```

Comment:
Do you expect changes? My point is to use an IEnumerable<ManifestEntry> or IReadOnlyList<ManifestEntry> instead